src/osemgrep/networking/Rule_fetching.ml

```diff
@@ -32,16 +32,11 @@ module XP = Xpattern
  * a pair, which would remove the need for partition_rules_and_errors.
  *)
 type rules_and_origin = {
-  origin : origin;
   rules : Rule.rules;
   errors : Rule.invalid_rule_error list;
+  origin : Parse_rule.origin; (* used by Validate_subcommand *)
```

Comment:
not sure it was good to move this type to Parse_rule.ml.
Maybe we should just pass a callback ?rewrite_rule_ids_hook:(Rule_id.t -> Rule_id.t) option
in Parse_rule.ml, that we set here in Rule_fetching.ml differently
when we were passed the --rewrite-rule-ids CLI flag.